js/src/admin/utils/ExtensionData.js

```diff
@@ -0,0 +1,170 @@
+import ItemList from '../../common/utils/ItemList';
+
+export default class ExtensionData {
```

Comment:
This class looks to me like an extender. Not sure if that's the intention, and I know we don't really use the JS extenders we have, but just an observation - not sure if that was intended.

Reply:
I think this is one of those places where having a pseudo-extender is simpler / safer than having users directly monkey-patch stuff on.

Reply:
I'm not saying to change it from that - I like having it like this instead.